urwid/display/_raw_display_base.py

```diff
@@ -654,22 +654,22 @@ def using_standout_or_underline(a):
                 if cs != "U":
                     run = run.translate(UNPRINTABLE_TRANS_TABLE)  # noqa: PLW2901
                 if first or lasta != a:
-                    o.append(attr_to_escape(a))
+                    output.append(attr_to_escape(a))
                     lasta = a
                 if first or lastcs != cs:
                     if cs not in {None, "0", "U"}:
                         raise ValueError(cs)
                     if lastcs == "U":
-                        o.append(escape.IBMPC_OFF)
+                        output.append(escape.IBMPC_OFF)
 
                     if cs is None:
-                        o.append(escape.SI)
+                        output.append(escape.SI)
                     elif cs == "U":
-                        o.append(escape.IBMPC_ON)
+                        output.append(escape.IBMPC_ON)
                     else:
-                        o.append(escape.SO)
+                        output.append(escape.SO)
                     lastcs = cs
-                o.append(run)
+                output.append(run.decode(util.get_encoding(), "replace"))
```

Comment:
Hmm, this code was originally working with bytes so that applications could be running with a non-utf-8 encoding without converting to Unicode and back (potentially losing data in the process) but I guess everything in raw display uses Unicode now?

Reply:
> Hmm, this code was originally working with bytes so that applications could be running with a non-utf-8 encoding without converting to Unicode and back (potentially losing data in the process) but I guess everything in raw display uses Unicode now?

Output in unicode was as minimum in version 2.0 (and even ignored target encoding): https://github.com/urwid/urwid/blob/release-2.0.0/urwid/raw_display.py#L848

Reply:
@wardi in python 3 we have STDOUT, which is always "Unicode". And it's not always buffered (so we cannot use buffer for output).